src/app/login/page.tsx

```diff
@@ -0,0 +1,247 @@
+          const errorData = res.error.data as { msg: string }
+          setErrorMessage(errorData.msg)
+          setOpenPopup(true)
+          setTitle('Error')
+        } else {
+          setErrorMessage('Unknown Error!')
+          setOpenPopup(true)
+          setTitle('Error')
+        }
+      }
+    })
+  }
+
+  const onSubmitRecPass: SubmitHandler<RecPassFormType> = async (data) => {
+    await sendEmail({ ...data }).then((res) => {
+      if (res) {
+        if ('data' in res) {
+          setMessage('An email has been sent to your email account.')
+          setOpenPrompt(true)
+          setTitle('Account Recovery')
+        } else if ('data' in res.error) {
+          const errorData = res.error.data as { msg: string }
+          setErrorMessage(errorData.msg)
+          setOpenPopup(true)
+          setTitle('Error')
+        } else {
+          setErrorMessage('Unknown Error!')
+          setOpenPopup(true)
+          setTitle('Error')
+        }
+      }
+    })
+  }
+
+  const handleClosePopup = () => {
+    setOpenPopup(false)
+  }
+
+  const handleRecoverPasswordClick = () => {
+    setOpen(true)
+  }
+
+  const handleCloseAccRecovery = () => {
+    setOpen(false)
+  }
+
+  return (
+    <Box
+      data-testid="login"
+      sx={{
+        display: 'flex',
+        flexDirection: { xs: 'column', md: 'row' },
+        minHeight: 'calc(100vh - 80px)',
+        width: '100%',
+      }}
+      borderRadius={'0 0 80% 0'}
+      alignItems={'center'}
+      justifyContent={'center'}
+      padding={{ xs: '4em 4em', lg: '4em 4em' }}
+    >
+      <Box sx={{ flex: 1, textAlign: 'left' }} data-testid="login-title">
+        <Typography variant="h3" fontWeight={'bold'}>
+          Log In to Plan Your Event
+        </Typography>
+        <Typography fontWeight={'bold'}>
+          If you do not have an account, you can{' '}
+          <Link style={{ color: 'teal' }} href="/register">
+            Register Here
+          </Link>
+        </Typography>
+      </Box>
+
+      <form
+        className="flex flex-col gap-3"
+        onSubmit={handleSubmit(onSubmit)}
+        style={{ flex: 1 }}
+        data-testid="login-form"
+      >
+        <Input
+          control={control}
+          name="username"
+          placeholder="Enter username"
+          required
+          data-testid="username-input"
+        />
+        <Input
+          type="password"
+          control={control}
+          name="password"
+          placeholder="Enter password"
+          required
+          data-testid="password-input"
+        />
+        <Typography
+          variant="body2"
+          align="right"
+          sx={{ textDecoration: 'underline', cursor: 'pointer' }}
+          onClick={handleRecoverPasswordClick}
+          data-testid="recover-account-link"
+        >
+          Recover Account
+        </Typography>
+        <button type="submit" className="bg-teal-50">
+          Log In
+        </button>
+      </form>
```

Comment:
I recommend refactoring your code by extracting the form into a separate component, for example, `LoginForm.tsx`. This approach aligns with the best practices of Next.js and enhances the modularity of your code.

By doing so, you adhere to the ‘Don’t Repeat Yourself’ (DRY) principle, which is a fundamental concept in software development aimed at reducing redundancy. This principle makes your code more maintainable and less prone to errors.

Even if this form component is used only once, separating it improves the readability of your code. It makes it easier for others to understand the structure and functionality of your application.

In summary, this change would make your code more organized, reusable, and easier to understand, which are all important qualities in effective software development.

Reply:
That's a really good point, by refactoring it will make the code look better and more readable.

Reply:
Besides the file mentioned earlier, I believe numerous other files could benefit from refactoring. Additionally, remember to set the target branch for merging to `staging`, not `main`! 😊